iml-manager-cli/src/api.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlManagerCliError;
+use console::Term;
+use structopt::{clap::arg_enum, StructOpt};
+
+arg_enum! {
+    #[derive(PartialEq, Debug)]
+    pub enum ApiType {
+        Delete,
+        Get,
+        Post,
+        Put,
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct ApiCommand {
+    #[structopt(possible_values = &ApiType::variants(), case_insensitive = true)]
+    call: ApiType,
+
+    path: String,
+
+    /// JSON formatted body to send
+    #[structopt(required_ifs(&[("call", "post"),("call", "put")]))]
+    body: Option<String>,
+}
+
+pub async fn api_cli(command: ApiCommand) -> Result<(), ImlManagerCliError> {
+    let term = Term::stdout();
+    let client = iml_manager_client::get_client()?;
+    let uri = iml_manager_client::create_api_url(command.path)?;
+    let body: Option<serde_json::Value> = command.body.map(|s| serde_json::from_str(&s).unwrap());
```

Comment:
oh, I missed double Option. `.and_then()` would be the best then :)

Reply:
Panic from a test tool if you use it wrong, I think, is okay.  Actually better than silently failing and setting body to `None`

Reply:
It doesn't really matter in this case, but https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.transpose would be useful to easily avoid the unwrap.